src/main/scala/scorex/core/network/RequestTracker.scala

```diff
@@ -47,18 +47,17 @@ class RequestTracker(
 
   private def sendTrackedRequest: Receive = {
     case message@SendToNetwork(m@Message(spec, _, _), strategy) if spec.messageCode == trackedRequestCode =>
-      askActor[Seq[ConnectedPeer]](networkControllerRef, GetConnectedPeers)
-        .map { peers =>
-          strategy.choose(peers.filter(_.peerInfo.exists(_.peerSpec.protocolVersion >= m.spec.protocolVersion)))
-            .foreach { peer =>
-              val requestKey = (trackedRequestCode, peer)
+      askActor[Seq[ConnectedPeer]](networkControllerRef, GetFilteredConnectedPeers(strategy, m.spec.protocolVersion))
+        .map {
+          _.foreach { peer =>
+            val requestKey = (trackedRequestCode, peer)
 
-              requestTracker += requestKey
+            requestTracker += requestKey
 
-              networkControllerRef ! message.copy(sendingStrategy = SendToPeer(peer))
+            networkControllerRef ! message.copy(sendingStrategy = SendToPeer(peer))
 
-              context.system.scheduler.scheduleOnce(deliveryTimeout, self, VerifyDelivery(requestKey, peer))
-            }
+            context.system.scheduler.scheduleOnce(deliveryTimeout, self, VerifyDelivery(requestKey, peer))
```

Comment:
as was discussed: account actual time used for ask message while schedule `deliveryTimeout`. Make `timeout` consistent to `deliveryTimeout`